zone/tradeskills.cpp

```diff
@@ -468,23 +468,26 @@ void Object::HandleCombine(Client* user, const NewCombine_Struct* in_combine, Ob
 		}
 	}
 
+	//pull the list of components so we can compare during lore check
+	const auto query = fmt::format("SELECT item_id "
```

Comment:
Instead of using this and passing the MySQL result object into another function, this is a perfect use case for using repositories. Please refactor to use 

There's already examples of them being used in tradeskills

```cpp
auto re = TradeskillRecipeEntriesRepository::GetWhere(
	database,
	fmt::format("recipe_id = {} ORDER BY id ASC", recipe_id)
);
```

Then you have a vector of rows that you can pass to the other function if you'd like and makes things far more flexible

Reply:
this is great feedback! made these changes.